### ISSUE TITLE
Issues with camera clipping in to geometry / objects, game breaking for many types of games.

### ISSUE THREAD

**Issue by xkamronx** [Sixze/ALS-Refactored#428]
Long story short - the camera being on the head bone allows the camera to clip through geo on basically any object fairly easily. The easiest method being crouching and looking straight at the ground while as close as possible to a mesh - the camera will clip through the geometry.

It is possible to partially remedy the issue by increasing the character's capsule size, but as noted in a previous interaction with you - I shouldn't be changing the capsule size as it doesn't interact / update the vaulting/mantling traces, so you can vault through areas you can't fall through.

Even with a larger capsule, the impulse on sprint can very easily also clip the character's head through a wall. In some instances, it's possible to get the camera through other objects like stairs / ramps by being underneath it, even in third person.

This can be pretty game breaking, where many games that have interaction can interact with objects through walls, ceilings, etc. I'm in the process of implementing additional checks, but I feel the general issue applies to all projects using Refactored, as the camera clipping in to geometry likely isn't a wanted feature in the vast majority of games.

**Comment by xkamronx** [comment]
To expand on this - there's also an issue with the crouched capsule height - the first person camera socket can very easily be outside of the collider, allowing players to see through slanted objects very easily. A lot of this is likely the animations themselves, but you can see very easily that the crouched height of the character doesn't match the half height of the capsule.

If the character fits nicely inside the standard capsule, their head sticks out of the half height.

**Comment by Mellnik** [comment]
I also noticed that camera clipping in first person is pretty much a deal breaker. I've tried improving it but I don't think I've fixed the source reason 🤷 

It would also be great to have an option to make the camera more "static". For example when you turn 180° while standing, the character turns around and the camera makes some weird movement which comes from the character's body movement. It's pretty annoying to play with.

**Comment by colorindarkness** [comment]
Have you tried adjusting the Camera Settings Data Asset?
![image](https://github.com/Sixze/ALS-Refactored/assets/9194221/2d57a599-a550-4806-a2e0-94ce4260b9ae)

**Comment by xkamronx** [comment]
> Have you tried adjusting the Camera Settings Data Asset? ![image](https://private-user-images.githubusercontent.com/9194221/300188848-2d57a599-a550-4806-a2e0-94ce4260b9ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0MDE0NTgsIm5iZiI6MTcwODQwMTE1OCwicGF0aCI6Ii85MTk0MjIxLzMwMDE4ODg0OC0yZDU3YTU5OS1hNTUwLTQ4MDYtYTJlMC05NGNlNDI2MGI5YWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjBUMDM1MjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDlkYmJhOWQ4ODgwYzI3YTg4MmQwM2M4MTA0Y2ZmYjc0MGFkYWNkYmEwNmQyN2FmNDgyNzk4ZmU4YzM0MmY4OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.GZOHfCcctyRSSDz70QxQQP8ukLiG-ClQmIkrOFmhnis)

Yep, there's not a end all be all fix with any settings configuration I have tested yet.

**Comment by xkamronx** [comment]
> I also noticed that camera clipping in first person is pretty much a deal breaker. I've tried improving it but I don't think I've fixed the source reason 🤷
> 
> It would also be great to have an option to make the camera more "static". For example when you turn 180° while standing, the character turns around and the camera makes some weird movement which comes from the character's body movement. It's pretty annoying to play with.

I haven't noticed much annoyance with the system overall but I do have a few of the built in movement features removed for users who said it was giving them motion sickness.

The camera clipping is a huge deal to me though. I increased the player's capsule size which I was told previously not to do because it messes with the vaulting trace, but it didn't seem like I had much of a choice for a somewhat close / simple fix to the camera being able to clip through walls basically on command.
